packages/frontend/src/utils/fiatValueManager.js

```diff
@@ -0,0 +1,118 @@
+import DataLoader from 'dataloader';
+import { Contract } from 'near-api-js';
+import Cache from 'node-cache';
+import { stringifyUrl } from 'query-string';
+
+import { REF_FINANCE_API_ENDPOINT, REF_FINANCE_CONTRACT} from '../config';
+import sendJson from '../tmp_fetch_send_json';
+import { wallet } from './wallet';
+
+const COINGECKO_PRICE_URL = 'https://api.coingecko.com/api/v3/simple/price';
+
+function wrapNodeCacheForDataloader(cache) {
+    return {
+        get: (...args) => {
+            return cache.get(...args);
+        },
+
+        set: (...args) => {
+            return cache.set(...args);
+        },
+
+        delete: (...args) => {
+            return cache.del(...args);
+        },
+
+        clear: (...args) => {
+            return cache.flushAll(...args);
+        }
+    };
+}
+
+export default class FiatValueManager {
+    constructor() {
+        this.coinGeckoFiatValueDataLoader = new DataLoader(
+            async (tokenIds) => {
+                try {
+                    const tokenFiatValues = await sendJson(
+                        'GET', 
+                        stringifyUrl({
+                            url: COINGECKO_PRICE_URL,
+                            query: {
+                                ids: tokenIds.join(','),
+                                vs_currencies: 'usd,eur,cny',
+                                include_last_updated_at: true
+                            }
+                        })
+                    );
+                    return tokenIds.map((id) => tokenFiatValues[id]);
+                } catch (error) {
+                    console.error(`Failed to fetch coingecko prices: ${error}`);
+                    // DataLoader must be constructed with a function which accepts 
+                    // Array<key> and returns Promise<Array<value>> of the same length
+                    // as the Array of keys
+                    return new Promise((resolve, reject) => {
+                        return [].fill({}, tokenIds.length);
+                    });
```

Comment:
`[].fill({}, tokenIds.length)` will always return an empty array AFAICT also, the promise does not resolve
```suggestion
                   return Promise.resolve(Array(tokenIds.length).fill({}));
```